kadmin/init.c

```diff
@@ -76,14 +71,14 @@ create_random_entry(krb5_principal princ,
 	mask |= KADM5_MAX_RLIFE;
     }
     ent.attributes |= attributes | KRB5_KDB_DISALLOW_ALL_TIX;
-    mask |= KADM5_ATTRIBUTES;
+    mask |= KADM5_ATTRIBUTES | KADM5_KEY_DATA;
 
-    /* Create the entry with a random password */
-    ret = kadm5_create_principal(kadm_handle, &ent, mask, password);
+    /* Create the entry with no keys or password */
+    ret = kadm5_s_create_principal_with_key(kadm_handle, &ent, mask);
```

Comment:
Does that actually randomize keys?  Somehow I can't tell from code inspection!

Reply:
`kadm5_randkey_principal() is called afterwards, before `KRB5_KDB_DISALLOW_ALL_TIX` is removed.